### PR TITLE
cleanup(ci): cleaned up circleci workflow.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area CI

**What this PR does / why we need it**:

Now that master and release CI are on github actions, cleanup up circle ci config.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: submodules/falcosecurity-rules
Comment (jasondellaluce) [inline submodules/falcosecurity-rules:1]: Is the submodule update wanted or a leftover?

Comment (LucaGuerra) [reply submodules/falcosecurity-rules:1]: I had the same question some time ago... I believe it is a leftover that is hard to remove.

**Comment by LucaGuerra** [comment]
Thank you @FedeDP for this 🙏

File: submodules/falcosecurity-rules
Comment (FedeDP) [reply submodules/falcosecurity-rules:1]: Exactly! I mean, i could rebase to remove it; is it a problem?

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by leogr — APPROVED** [review]
:partying_face:

**Pull request merged**